valen-ui/src/main/java/be/twofold/valen/ui/viewers/data/DataViewer.java

```diff
@@ -0,0 +1,156 @@
+package be.twofold.valen.ui.viewers.data;
+
+import be.twofold.valen.core.game.*;
+import be.twofold.valen.ui.viewers.*;
+import javafx.collections.*;
+import javafx.scene.*;
+import javafx.scene.control.*;
+
+import java.lang.reflect.*;
+import java.util.*;
+
+public class DataViewer extends TreeView<DataViewer.PreviewItem> implements Viewer {
+    public DataViewer() {
+        setCellFactory(p -> new PreviewItemTreeCellImpl());
+    }
+
+    @Override
+    public boolean canPreview(AssetType type) {
+        return type == AssetType.Image || type == AssetType.Text || type == AssetType.Model; //TODO: Add other supported types
+    }
+
+    @Override
+    public void setData(Object data) {
+        if (data.getClass().isRecord()) {
+            var rootItem = new PreviewValueTreeItem(new PreviewItem(data.getClass().getSimpleName(), data));
+            rootItem.setExpanded(true);
+            setRoot(rootItem);
+        } else if (data instanceof Map<?, ?> map) {
+            var rootItem = new PreviewValueTreeItem(new PreviewItem(data.getClass().getSimpleName(), map));
+            rootItem.setExpanded(true);
+            setRoot(rootItem);
+        } else {
+            setRoot(null);
+        }
+    }
+
+    @Override
+    public Node getNode() {
+        return this;
+    }
+
+    @Override
+    public String getName() {
+        return "Data";
+    }
+
+    record PreviewItem(String name, Object value) {
+    }
+
+    static class PreviewValueTreeItem extends TreeItem<PreviewItem> {
```

Comment:
smol nitpick, extract these to a separate file, same with PreviewItem and PreviewItemTreeCellImpl